### PR TITLE
feat: Login Page - State 2

### LINKED ISSUE #868: Build Login Page - State 2
A new NextJS page will be created that handles a custom implementation of Next Auth v4.  The page will be located at ‘{base_url}/login’ and completes both states of the login process.

This state is part of a larger [Login Flow](https://docs.google.com/document/d/1ua3QfIHAcZwwodgGN2alkVZa5brgHr-P0Mw85gujrz8/edit?lfhs=2#heading=h.6ox7ti5fg93o).  Please review the complete flow before working on this page.  

[Login Page State 1 Issue](https://app.zenhub.com/workspaces/gcforms-60cb8929764d71000e481cab/issues/cds-snc/platform-forms-client/867)

Once a success (200 http code) response is received from `{base_url}/api/token/temporary` the state of the login page will change and display the second state of the login process.  This state is local to the user (client side) and does not need to be stored server side. In case that the Temporary Token is not received by the Program Administrator there should be a method that allows the user to return to the previous state.  For more information see [State 2 page design](https://www.figma.com/file/ZPt6quJaJ2bVYZboCccWJs/Retrieval-UI?node-id=637%3A950).

The second state of the Login process will leverage a [custom credential provider](https://next-auth.js.org/providers/credentials) that verifies existing credentials stored in the Form Users table.  Existing logic that verifies the Temporary Token when submitted to the `{base_url}/api/id/{formId}/retrieval` API has potential to be reused for this purpose.  On the login page a form will accept a temporary token and once submitted leverage the [signIn()](https://next-auth.js.org/getting-started/client#signin) function available on the Client API of Next Auth.  An example can be found here: https://next-auth.js.org/configuration/pages#credentials-sign-in.  The result of this sign in function will be Next Auth creating and sending a JWT to the browser for session authentication.

## Design for Login State 2:
- [Form](https://www.figma.com/file/ZPt6quJaJ2bVYZboCccWJs/Retrieval-UI?node-id=637%3A950)
- [Error State](https://www.figma.com/file/ZPt6quJaJ2bVYZboCccWJs/Retrieval-UI?node-id=741%3A1310)

## Acceptance Criteria:
- [x] State 2 is a separate component from the Login Page itself
- [ ] Component follows design as presented in Figma
- [ ] Component is accessible
- [ ] Component handles error state on incorrect credentials
- [ ] Component redirects to Acceptable Use Page (`{base_url}/auth/policy`)
- [ ] Component can switch back to State 1 through user action.

### PR DESCRIPTION
**Pull request by jeberhardt**
# Summary | Résumé

Closes #868 

| Before                                          | After                                        |
| ----------------------------------------------- | -------------------------------------------- |
| Insert Image showing before the change was made | 
Screen Shot 2022-08-08 at 5 27 45 PM
 |

# Test instructions | Instructions pour tester la modification

- go to `~/auth/login`
- enter the email and bearerToken for a form you've got access to
- the page will refresh to show this content for the temporary token
- check your email and enter the temporary token onto this page

# Unresolved questions / Out of scope | Questions non résolues ou hors sujet

Are there any related issues or tangent features you consider out of scope for
this issue that could be addressed in the future? If so please create issues and provide
links in this section

# Pull Request Checklist

Please complete the following items in the checklist before you request a review:

- [x] Have you completely tested the functionality of change introduced in this PR? Is the PR solving the problem it's meant to solve within the scope of the related issue?
- [x] The PR does not introduce any new issues such as failed tests, console warnings or new bugs.
- [ ] If this PR adds a package have you ensured its licensed correctly and does not add additional security issues?
- [x] Is the code clean, readable and maintainable? Is it easy to understand and comprehend.
- [ ] Does your code have adequate comprehensible comments? Do new functions have [docstrings](https://tsdoc.org/)?
- [x] Have you modified the change log and updated any relevant documentation?
- [x] Is there adequate test coverage? Both unit tests and end-to-end tests where applicable?
- [x] If your PR is touching any UI is it accessible? Have you tested it with a screen reader? Have you tested it with automated testing tools such as axe?

### CODE REVIEW

File: pages/api/token/temporary.ts [hunk 1]
```diff
@@ -37,7 +37,7 @@ const handler = async (req: NextApiRequest, res: NextApiResponse, { formID }: Mi
     await sendTemporaryTokenByEmail(email, temporaryToken);
 
     logMessage.info(`Temporary Token Requested: Form ID: ${formID} Email: ${email}`);
-    res.status(200).json({ message: "success" });
+    res.status(200).json({ message: "success", formID: formID });
```
Comment (craigzour) [inline pages/api/token/temporary.ts:40]: I can't see where you need the `formID` in the rest of the pull request.

File: pages/auth/login.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+import TemporaryToken from "@components/login/TemporaryToken";
+import { useTranslation } from "next-i18next";
+import { GetServerSideProps } from "next";
+
+export interface LoginStageProps {
+  setParentStage: (stage: number) => void;
+}
+
+const Login = () => {
+  const { t } = useTranslation("login");
```
Comment (craigzour) [inline pages/auth/login.tsx:13]: Looks like the translation object here is not used.

File: pages/api/token/temporary.ts (hunk 1 above)
Comment (jeberhardt) [reply pages/api/token/temporary.ts:40]: @craigzour  - You're right; this was put in before I figured out what Bryan had put together for the Credentials Provider, and is not a part of this PR.

**Review by craigzour — APPROVED** [review]

**Review by falila — DISMISSED** [review]
Nice work!🍉

**Review by craigzour — APPROVED** [review]

**Review by falila — DISMISSED** [review]

**Review by craigzour — APPROVED** [review]

**Pull request merged**